datasets/cmu_hinglish_dog/cmu_hinglish_dog.py

```diff
@@ -0,0 +1,190 @@
+                "history": datasets.Sequence(
+                    {
+                        "docIdx": datasets.Value("int64"),
+                        "text": datasets.Value("string"),
+                        "uid": datasets.Value("string"),
+                        "utcTimestamp": datasets.Value("string"),
+                    }
+                ),
+                "rating": datasets.Value("int64"),
+                "status": datasets.Value("int64"),
+                "uid1LogInTime": datasets.Value("string"),
+                "uid1LogOutTime": datasets.Value("string"),
+                "uid1response": datasets.Sequence(
+                    {
+                        "feedback": datasets.Value("string"),
+                        "response": datasets.Sequence([datasets.Value("int64")]),
+                        "type": datasets.Value("string"),
+                    }
+                ),
+                "uid2response": datasets.Sequence(
+                    {
+                        "feedback": datasets.Value("string"),
+                        "response": datasets.Sequence([datasets.Value("int64")]),
+                        "type": datasets.Value("string"),
+                    }
+                ),
+                "user2_id": datasets.Value("string"),
+                "whoSawDoc": datasets.Sequence([datasets.Value("string")]),
+                "wikiDocumentIdx": datasets.Value("int64"),
+            }
+        )
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """The linking part between Hinglish data and English data is inspired from the implementation in GLUECoS.
+        Refer here for the original script https://github.com/microsoft/GLUECoS/blob/7fdc51653e37a32aee17505c47b7d1da364fa77e/Data/Preprocess_Scripts/preprocess_mt_en_hi.py"""
+
+        eng_path = dl_manager.download_and_extract(_URL_ENGLISH)
+
+        # WikiData file is not required for this
+        if os.path.isdir(os.path.join(eng_path, "datasets-CMU_DoG-master", "WikiData")):
+            shutil.rmtree(os.path.join(eng_path, "datasets-CMU_DoG-master", "WikiData"))
+
+        data_dir_en = os.path.join(eng_path, "datasets-CMU_DoG-master", "Conversations")
+
+        hi_en_path = dl_manager.download_and_extract(_URL_HINGLISH)
+        data_dir_hi_en = os.path.join(
+            hi_en_path, "CMUHinglishDoG", "Conversations_Hinglish"
+        )
+
+        # Extensions in Hinglish dataset are .json_1 and .json_2, converting them to .json
+        for split in os.listdir(data_dir_hi_en):
+            for f in os.listdir(os.path.join(data_dir_hi_en, split)):
+                f = os.path.join(os.path.join(data_dir_hi_en, split, f))
+                p = Path(f)
+                os.rename(f, p.with_suffix(".json"))
```

Comment:
You can remove this part an open the .json_1 and .json_2 files

Reply:
Yes ``.json_1`` and ``.json_2`` can be opened, but the dummy data generation script skips it because the extension is not ``.json`` [here](https://github.com/huggingface/datasets/blob/b6bfe88d9ae11ba51eb8b4ba720826cf9274969c/src/datasets/commands/dummy_data.py#L136)

Reply:
Also I tested out without ``--json_field`` also, it gives the error Can't parse output
```
Using custom data configuration default
Couldn't generate dummy file 'datasets/cmu_hinglish_dog/dummy/0.0.0/dummy_data/Conversations.zip/datasets-CMU_DoG-master/README.md'. Ignore that if this file is not useful for dummy data.
Traceback (most recent call last):
  File "/home/ishan/anaconda3/bin/datasets-cli", line 8, in <module>
    sys.exit(main())
  File "/home/ishan/anaconda3/lib/python3.8/site-packages/datasets/commands/datasets_cli.py", line 33, in main
    service.run()
  File "/home/ishan/anaconda3/lib/python3.8/site-packages/datasets/commands/dummy_data.py", line 318, in run
    self._autogenerate_dummy_data(
  File "/home/ishan/anaconda3/lib/python3.8/site-packages/datasets/commands/dummy_data.py", line 346, in _autogenerate_dummy_data
    dl_manager.auto_generate_dummy_data_folder(
  File "/home/ishan/anaconda3/lib/python3.8/site-packages/datasets/commands/dummy_data.py", line 87, in auto_generate_dummy_data_folder
    total += self._create_dummy_data(
  File "/home/ishan/anaconda3/lib/python3.8/site-packages/datasets/commands/dummy_data.py", line 178, in _create_dummy_data
    total += self._create_dummy_data(
  File "/home/ishan/anaconda3/lib/python3.8/site-packages/datasets/commands/dummy_data.py", line 145, in _create_dummy_data
    raise ValueError(
ValueError: Couldn't parse columns ['date', 'history', 'rating', 'status', 'uid1LogInTime', 'uid1LogOutTime', 'uid1response', 'uid2LogInTime', 'uid2LogOutTime', 'uid2response', 'user1_id', 'user2_id', 'whoSawDoc', 'wikiDocumentIdx']. Maybe specify which json field must be used to read the data with --json_field <my_field>.
```

Reply:
[This line](https://github.com/huggingface/datasets/blob/b6bfe88d9ae11ba51eb8b4ba720826cf9274969c/src/datasets/commands/dummy_data.py#L142) requires every value of json to a be a ``list``. That is not true in this data since each datapoint is a seperate json, so do I need to pre-process the data?

Reply:
Good catch, that's indeed an issue for your case. I guess this format is not supported for the dummy data generator. You can still create the dummy data manually.

To do so:
1. go into the directory `datasets/cmu_hinglish_dog/dummy/0.0.0` and create a directory `dummy_data`.
2. go into this directory and create two directories: `Conversations.zip`and `CMUHinglishDoG.zip` (don't zip them, this is just to say that they contains the dummy data for each archive)
3. copy the content of the actual archives in them, but only keep 1 or 2 files per split. You can also remove the unnecessary files like the wikidata folder
4. zip the dummy_data directory into `dummy_data.zip`
5. run the command
```bash
RUN_SLOW=1 pytest tests/test_dataset_common.py::LocalDatasetTest::test_load_dataset_all_configs_cmu_hinglish_dog
```

Let me know if that worked, and feel free to ping me if I can help

Reply:
Okay sure, I'll do it using these steps. Thanks!